### PR TITLE
Week 11 – Happy Thoughts Project

### PR DESCRIPTION
**Pull request by emmajosefina**
Link to my deployed project: https://happy-thoughts-today.netlify.app/

My reflections on how this week's project turned out: Very well!

Things I'd like to have clarified or explained in more detail: No, I don't think so :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
-Replace this readme with your own information about your project.
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+I created a Happy Thoughts-app, where I practiced my React state skills by fetching and posting data to API.
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+My main problem this week was to figure out the best way how to structure my code as clean as possible throught components in React.
```
Comment (jiiinCho) [inline README.md:7]: What a great mindset ❤️

File: code/src/components/Footer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+
+const Footer = () => {
+
+return (
+
+
+    <div className="footer-section">
+       <p> Made with <span role="img" aria-label="heart emoji">❤️</span> by <a className="footer-link" href="https://github.com/emmajosefina/" target="_blank" rel="noopener noreferrer">Emma</a></p>
```
Comment (jiiinCho) [inline code/src/components/Footer.js:10]: Oh, lovely footer!!

File: code/src/components/HappyThoughtCards.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        .then(thoughts => setThoughts(thoughts))
+}, [])
+
+
+// Post likes to the API
+const onLike = (thoughtId) => {
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${thoughtId}/like`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json'},
+      body: ''
```
Comment (jiiinCho) [inline code/src/components/HappyThoughtCards.js:21]: you may not need line 21 if the request does not have any body data ! :)

File: code/src/components/HappyThoughtCards.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }).then(() => updateLikes(thoughtId))
+  };
+
+
+// Function for updating likes
+const updateLikes = (thoughtId) => {
```
Comment (jiiinCho) [inline code/src/components/HappyThoughtCards.js:27]: great function naming! easy to follow !

File: code/src/components/HappyThoughtCards.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        <section className="thought-box" key={thought._id} thought={thought} onLike={onLike}>
+                <p className="message">
+                    {thought.message}
+                    
+                </p>
+
+
+            <div className="likes-container">
+                <button 
+                className={(thought.hearts === 0 ? "heart-btn" : "heart-btn red-heart-btn")}
```
Comment (jiiinCho) [inline code/src/components/HappyThoughtCards.js:56]: great ternary operator!

File: code/src/components/ThoughtForm.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    event.preventDefault()
+
+
+    fetch("https://happy-thoughts-technigo.herokuapp.com/thoughts", {
+        method: "POST",
+        headers: {'Content-Type': "application/json"},
+        body: JSON.stringify({ message: newThought })
+})
+    .then((res) => res.json())
+    .then((newThought) => props.setThoughts((thoughts) => [newThought, ...thoughts]))
```
Comment (jiiinCho) [inline code/src/components/ThoughtForm.js:21]: great technique to add and sort the item!

File: code/src/components/ThoughtForm.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                 value={newThought}
+                 onChange={event => setNewThought(event.target.value)}
+                 >
+
+                </textarea>
+
+            <div className="input-container-details">
+
+                <div>
+                    <button disabled={newThought.length < 6 || newThought.length > 140 ? true : false} /* If input is less than 6 or longer than 140 characters, display button will be disabled */
```
Comment (jiiinCho) [inline code/src/components/ThoughtForm.js:53]: nice comment and script! so advanced! I didn't know there is an attribute for disabled option in button element!

File: code/src/index.css [hunk 8]
```diff
@@ -1,13 +1,183 @@
...
+  color: var(--clr-grey);
+  font-size: 14px;
+  display: flex;
+  align-items: center;
+  margin: 0 0.5em;
+  justify-content: center;
+}
+
+.heart-btn {
+  background: #eaeaea;
```
Comment (jiiinCho) [inline code/src/index.css:148]: maybe include this as variable in the :root?

**Review by jiiinCho — COMMENTED** [review]
Oh great script you wrote Emma! I am so impressed! You tried many advanced JS techniques and your web application works smoothly without any errors/warnings/issues :) super great job! (Sorry for late review!!)

File: code/src/components/HappyThoughtCards.js (hunk 3 above)
Comment (emmajosefina) [reply code/src/components/HappyThoughtCards.js:21]: Thank you, had no idea! 💯

**Pull request closed without merging**